### ISSUE TITLE
`strongifiedSelf` strips some necessary backticks

### ISSUE THREAD

**Issue by heiberg** [nicklockwood/SwiftFormat#1670]
# Description

Swift 5.10 (Xcode 15.3) introduced the `nonisolated(unsafe)` keyword.

In the following line the backticks are required:

```swift
nonisolated(unsafe) let `self` = self
```

The `strongifiedSelf` rule strips these, which results in the error

> 🔴 Keyword 'self' cannot be used as an identifier here

Note that the example here is not strongifying a weak `self`, but rather declaring a non-isolated version of `self` ([useful in certain test cases](https://forums.swift.org/t/swift-5-10-concurrency-and-xctest/69929/8)) .

# Environment

* Xcode 15.3
* Swift 5.10
* SwiftFormat 0.53.5

**Comment by nicklockwood** [comment]
@heiberg I've not been able to reproduce this in 0.53.5 or 0.53.6. It's possible that the bug depends on some surrounding context - could you provide the whole method? (or whatever minimal code is needed to reproduce the problem in isolation)

**Comment by heiberg** [comment]
Sure! Sorry for the delay, I missed the notification here.

## Input

```swift
class Hello {
    func foo() {
        nonisolated(unsafe) let `self` = self
    }
}
```

## Command

```
swiftformat --swiftversion 5.10 Input.swift
```

## Expected Output

Unchanged.

## Observed Output

```swift
class Hello {
    func foo() {
        nonisolated(unsafe) let self = self
    }
}
```

Unexpected change:

```diff
class Hello {
    func foo() {
-        nonisolated(unsafe) let `self` = self
+        nonisolated(unsafe) let self = self
    }
}
```

## Environment

SwiftFormat 0.53.5

**Comment by heiberg** [comment]
@nicklockwood Was this issue reproducible given the above?

**Comment by nicklockwood** [comment]
@heiberg sorry, I also missed this update. I believe the issue was that it's the `strongifiedSelf` rule rather than `redundantBackticks` that was causing the issue. I should have a fix shortly.

**Comment by nicklockwood** [comment]
@heiberg fixed in 0.53.10
